src/classes/CON_DoNotContact_TDTM.cls

```diff
@@ -32,95 +32,177 @@
+            }
 
-        con.HasOptedOutOfEmail = con.Do_Not_Contact__c;
-        con.DoNotCall = con.Do_Not_Contact__c;
-        con.HasOptedOutOfFax  = con.Do_Not_Contact__c;
+            if (con.Do_Not_Contact__c != oldCon.Do_Not_Contact__c){
+                doNotContactContactsList.add(con);
+            }
+        }
 
+        if (deceasedContactsList.isEmpty() == false){
+            this.syncNamingExclusionsOnContacts(deceasedContactsList);
+            this.syncDoNotContactFieldsOnContacts(deceasedContactsList);
+        }
+
+        if (doNotContactContactsList.isEmpty() == false){
+            this.syncDoNotContactFieldsOnContacts(doNotContactContactsList);
+        }
     }
 
-    /*******************************************************************************************************
-    * @description Copies the values from the naming exclusions checkboxes to the multiselect picklist.
-    * @param con The Contact
-    * @return void
-    ********************************************************************************************************/
-    public static void copyNamingExclusionsFromCheckboxes(Contact con) {
-        string mspValue = '';
+    /**
+     * @description Synchronizes the communication preference fields
+     * (Do not Call, Email Opt Out, Fax Opt Out) on a contact record
+     * to match the status specified by the "Do Not Contact" field.
+     * @param contactsList List of contacts to synchronize
+     */ 
+    @TestVisible
+    private void syncDoNotContactFieldsOnContacts(List<Contact> contactsList){
+        for (Contact con : contactsList){
+            con.HasOptedOutOfEmail = con.Do_Not_Contact__c;
+            con.DoNotCall = con.Do_Not_Contact__c;
+            con.HasOptedOutOfFax  = con.Do_Not_Contact__c;
+        }
+    }
 
-        if (con.Exclude_from_Household_Name__c)
-            mspValue += 'Household__c.Name;';
-        if (con.Exclude_from_Household_Formal_Greeting__c)
-            mspValue += 'Household__c.Formal_Greeting__c;';
-        if (con.Exclude_from_Household_Informal_Greeting__c)
-            mspValue += 'Household__c.Informal_Greeting__c;';
+    /**
+     * @description Synchronizes the Household Naming Exclusion fields 
+     * (checkboxes and picklist) on a contact record to match the status 
+     * specified by the "Deceased" field.
+     * @param contactsList List of contacts to synchronize
+     */ 
+    @TestVisible
+    private void syncNamingExclusionsOnContacts(List<Contact> contactsList){
+        for (Contact con : contactsList){
+            con.Exclude_from_Household_Name__c = con.Deceased__c;
+            con.Exclude_from_Household_Formal_Greeting__c = con.Deceased__c;
+            con.Exclude_from_Household_Informal_Greeting__c = con.Deceased__c;
+        }
 
-        if (mspValue.endsWith(';'))
-            mspValue = mspValue.substring(0,mspValue.length()-1);
+        this.syncHHNamingExclusionsPicklistFromCheckboxes(contactsList);
```

Comment:
Looks like we have a chain of methods here. Shall we try to de-couple them since you can just call syncHHNamingExclusionsPicklistFromCheckboxes in 'process' methods?